Sources/SwiftKafka/Configuration/KafkaConfiguration.swift

```diff
@@ -0,0 +1,198 @@
+        public var keepaliveEnable: Bool = false
+
+        /// Disable the Nagle algorithm (TCP_NODELAY) on broker sockets.
+        public var nagleDisable: Bool = false
+
+        /// Disconnect from broker when this number of send failures (e.g., timed out requests) is reached. Disable with 0. WARNING: It is highly recommended to leave this setting at its default value of 1 to avoid the client and broker to become desynchronized in case of request timeouts. NOTE: The connection is automatically re-established.
+        public var maxFails: Int = 1
+
+        /// Maximum time allowed for broker connection setup (TCP connection setup as well SSL and SASL handshake). If the connection to the broker is not fully functional after this the connection will be closed and retried.
+        public var connectionSetupTimeoutMilliseconds: Int = 30000
+    }
+
+    /// Broker options.
+    public struct BrokerOptions: Sendable, Hashable {
+        /// How long to cache the broker address resolving results (milliseconds).
+        public var addressTTL: Int = 1000
+
+        /// Allowed broker ``KafkaConfiguration/IPAddressFamily``.
+        public var addressFamily: KafkaConfiguration.IPAddressFamily = .any
+    }
+
+    /// Reconnect options.
+    public struct ReconnectOptions: Sendable, Hashable {
+        /// The initial time to wait before reconnecting to a broker after the connection has been closed. The time is increased exponentially until reconnect.backoff.max.ms is reached. -25% to +50% jitter is applied to each reconnect backoff. A value of 0 disables the backoff and reconnects immediately.
+        public var backoffMilliseconds: Int = 100
+
+        /// The maximum time to wait before reconnecting to a broker after the connection has been closed.
+        public var backoffMaxMilliseconds: Int = 10000
+    }
+
+    /// SSL options.
+    public struct SSLOptions: Sendable, Hashable {
+        /// Path to client's private key (PEM) used for authentication.
+        public var keyLocation: String = ""
+
+        /// Private key passphrase (for use with ssl.key.location).
+        public var keyPassword: String = ""
+
+        /// Path to client's public key (PEM) used for authentication.
+        public var certificateLocation: String = ""
+
+        /// File or directory path to CA certificate(s) for verifying the broker's key. Defaults: On Windows the system's CA certificates are automatically looked up in the Windows Root certificate store. On Mac OSX this configuration defaults to probe. It is recommended to install openssl using Homebrew, to provide CA certificates. On Linux install the distribution's ca-certificates package. If OpenSSL is statically linked or ssl.ca.location is set to probe a list of standard paths will be probed and the first one found will be used as the default CA certificate location path. If OpenSSL is dynamically linked the OpenSSL library's default path will be used (see OPENSSLDIR in openssl version -a).
```

Comment:
Because I'm a pedant:

```suggestion
        /// File or directory path to CA certificate(s) for verifying the broker's key. Defaults: On Windows the system's CA certificates are automatically looked up in the Windows Root certificate store. On macOS this configuration defaults to probe. It is recommended to install openssl using Homebrew, to provide CA certificates. On Linux install the distribution's ca-certificates package. If OpenSSL is statically linked or ssl.ca.location is set to probe a list of standard paths will be probed and the first one found will be used as the default CA certificate location path. If OpenSSL is dynamically linked the OpenSSL library's default path will be used (see OPENSSLDIR in openssl version -a).
```